fix: match semver patch

ivanstanev opened the pull request:
A previous commit introduced semver.coerce() on the patch version range, when doing 'snyk protect'.
The problem is that coerce() returns a SemVer type, not a string. The existing code uses semver.satisfies() to check if the given patch applies to the node_module, but since satisfies() expects a string, the patch was not applied since the check never passed.

This resulted in patches being skipped, but we marked them as applied.

The following fix restores the previous functionality of running satisfies() on the patch range but also keeps the existing check of using coerce() while also fixing the output of coerce() to be a string (by reading the version property of the SemVer type). This way we do a check that the patch version is a SemVer range, but we also correctly cover the case where it's not a SemVer range (which was originally the intention).

- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team


#### What are the relevant tickets?
[Zendesk Ticket 1968](https://snyk.zendesk.com/agent/tickets/1968)

gitphill commented on test/protect-semver-patch.test.ts:
```diff
@@ -0,0 +1,90 @@
+// tslint:disable-next-line: no-var-requires
+const debug = require('debug')('snyk');
```
you can do the following to avoid tslint:disable-next-line

```
import * as debugModule from 'debug';
const debug = debugModule('snyk');
```

Similar on line 6 too

ivanstanev replied:
I'm requiring a .json file though, I get an error when I try to use TS import 🤔

ivanstanev commented:
@gitphill I've updated the PR!
Please note this change: https://github.com/snyk/snyk/pull/825/files#diff-d97913173ab0f19dd07e005afe851b7aR8-R15
I have added the same version check here because this code path runs before the actual apply-patch. This code path filters out patches that _can_ be applied, so it wouldn't make sense to have different checks at different points in time.

orsagie commented on test/fixtures/hoek@4.2.0-nopatch.json:
```diff
@@ -0,0 +1,178 @@
+{
```
These files should not be checked in.

orsagie replied:
Once you remove them, tests should pass

gitphill replied:
Not sure that's true @orsagie - this file is used to test when no patch is available, see https://github.com/snyk/snyk/blob/4bb172caa4232565e979f1d19b38adb46e8dd54c/test/protect-semver-patch.test.ts#L68

The failing CI tests are unrelated to this change and I've seen them fail regularly in the past

snyksec commented:
:tada: This PR is included in version 1.237.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/snyk)
- [GitHub release](https://github.com/snyk/snyk/releases/tag/v1.237.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
